File: Unknownbank/app/src/main/java/com/kasatria/kasatriaunknownbank/CreditCardReviewScreen.kt
package com.kasatria.kasatriaunknownbank

import androidx.compose.foundation.Image
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.navigationBarsPadding
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.statusBarsPadding
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowBack
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

import com.kasatria.kasatriaunknownbank.ui.theme.PrimaryBlue
import com.kasatria.kasatriaunknownbank.ui.theme.TextPrimary
import com.kasatria.kasatriaunknownbank.ui.theme.White


@Composable
fun CreditCardReviewScreen(
    productName: String,
    applicationData: CreditCardApplicationData,
    onBack: () -> Unit,
    onEditPersonal: () -> Unit,
    onNext: () -> Unit,
    onEditJob: () -> Unit,
    onEditExtraDetails: () -> Unit
) {

    Box(
        modifier = Modifier.fillMaxSize()
    ) {

        /*
         * Decorative background
         */

        Image(
            painter = painterResource(
                id = R.drawable.full_background_light
            ),
            contentDescription = null,
            modifier = Modifier.fillMaxSize(),
            contentScale = ContentScale.Crop
        )


        Column(
            modifier = Modifier
                .fillMaxSize()
                .statusBarsPadding()
        ) {

            /*
             * =================================
             * HEADER
             * =================================
             */

            Box(
                modifier = Modifier
                    .fillMaxWidth()
                    .height(56.dp)
            ) {

                IconButton(
                    onClick = onBack,

                    modifier = Modifier
                        .padding(start = 8.dp)
                        .align(
                            Alignment.CenterStart
                        )
                ) {

                    Icon(
                        imageVector =
                            Icons.AutoMirrored.Filled.ArrowBack,

                        contentDescription = "Back",

                        tint = TextPrimary
                    )
                }


                Text(
                    text =
                        "Applying for $productName",

                    modifier = Modifier
                        .align(Alignment.Center)
                        .padding(
                            start = 55.dp,
                            end = 30.dp
                        ),

                    color = Color(0xFF333333),

                    fontSize = 13.sp,

                    fontWeight =
                        FontWeight.Bold,

                    maxLines = 1
                )
            }


            /*
             * =================================
             * PROGRESS
             * =================================
             */

            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(
                        start = 20.dp,
                        end = 23.dp,
                        top = 20.dp
                    ),

                horizontalArrangement =
                    Arrangement.SpaceBetween,

                verticalAlignment =
                    Alignment.CenterVertically
            ) {

                Text(
                    text = "Review Details",

                    color = Color(0xFF333333),

                    fontSize = 15.sp,

                    fontWeight =
                        FontWeight.Bold
                )


                Text(
                    text = "100%",

                    color =
                        Color(0xFF0EAF5F),

                    fontSize = 14.sp,

                    fontWeight =
                        FontWeight.Medium
                )
            }


            Spacer(
                modifier =
                    Modifier.height(11.dp)
            )


            /*
             * Full progress bar
             */

            Box(
                modifier = Modifier
                    .padding(horizontal = 20.dp)
                    .fillMaxWidth()
                    .height(6.dp)
                    .background(
                        color =
                            Color(0xFF0EAF5F),

                        shape =
                            RoundedCornerShape(100.dp)
                    )
            )


            Spacer(
                modifier =
                    Modifier.height(26.dp)
            )


            /*
             * =================================
             * REVIEW CONTENT
             * =================================
             */

            Column(
                modifier = Modifier
                    .weight(1f)
                    .verticalScroll(
                        rememberScrollState()
                    )
                    .padding(
                        start = 20.dp,
                        end = 20.dp,
                        bottom = 110.dp
                    )
            ) {

                PersonalDetailsReviewCard(
                    applicationData = applicationData,
                    onEdit = onEditPersonal,

                )

                Spacer(
                    modifier = Modifier.height(16.dp)
                )

                JobDetailsReviewCard(
                    applicationData = applicationData,
                    onEdit = onEditJob
                )

                Spacer(
                    modifier = Modifier.height(16.dp)
                )

                ExtraDetailsReviewCard(
                    applicationData = applicationData,
                    onEdit = onEditExtraDetails
                )
            }
        }


        /*
         * =================================
         * NEXT BUTTON
         * =================================
         */

        Button(
            onClick = onNext,

            modifier = Modifier
                .align(
                    Alignment.BottomCenter
                )
                .navigationBarsPadding()
                .padding(
                    start = 16.dp,
                    end = 16.dp,
                    bottom = 16.dp
                )
                .fillMaxWidth()
                .height(46.dp),

            shape =
                RoundedCornerShape(100.dp),

            colors =
                ButtonDefaults.buttonColors(
                    containerColor =
                        PrimaryBlue,

                    contentColor =
                        White
                )
        ) {

            Text(
                text = "Next",

                fontSize = 15.sp,

                fontWeight =
                    FontWeight.Bold
            )
        }
    }
}


@Composable
private fun PersonalDetailsReviewCard(
    applicationData: CreditCardApplicationData,
    onEdit: () -> Unit
) {

    val shape =
        RoundedCornerShape(12.dp)

    Column(
        modifier = Modifier
            .fillMaxWidth()
            .shadow(
                elevation = 3.dp,
                shape = shape
            )
            .background(
                color = White,
                shape = shape
            )
            .border(
                width = 1.dp,
                color = Color(0xFFEEEEEE),
                shape = shape
            )
            .padding(17.dp)
    ) {

        /*
         * Card header
         */

        Row(
            modifier =
                Modifier.fillMaxWidth(),

            verticalAlignment =
                Alignment.CenterVertically,

            horizontalArrangement =
                Arrangement.SpaceBetween
        ) {

            Row(
                verticalAlignment =
                    Alignment.CenterVertically
            ) {

                Image(
                    painter = painterResource(
                        id = R.drawable.review_person
                    ),

                    contentDescription = null,

                    modifier =
                        Modifier.size(16.dp)
                )


                Spacer(
                    modifier =
                        Modifier.size(8.dp)
                )


                Text(
                    text = "Personal Details",

                    color = Color.Black,

                    fontSize = 14.sp,

                    fontWeight =
                        FontWeight.Bold
                )
            }


            Row(
                modifier =
                    Modifier.clickable {
                        onEdit()
                    },

                verticalAlignment =
                    Alignment.CenterVertically
            ) {

                Text(
                    text = "EDIT",

                    color = PrimaryBlue,

                    fontSize = 12.sp,

                    letterSpacing = 0.6.sp
                )


                Spacer(
                    modifier =
                        Modifier.size(4.dp)
                )


                Image(
                    painter = painterResource(
                        id = R.drawable.review_edit
                    ),

                    contentDescription =
                        "Edit personal details",

                    modifier =
                        Modifier.size(12.dp)
                )
            }
        }


        Spacer(
            modifier =
                Modifier.height(9.dp)
        )


        Box(
            modifier = Modifier
                .fillMaxWidth()
                .height(1.dp)
                .background(
                    Color(0xFFF2F4F6)
                )
        )


        Spacer(
            modifier =
                Modifier.height(8.dp)
        )


        /*
         * Details
         */

        ReviewDetail(
            label = "Title",
            value = applicationData.title
        )

        ReviewDetail(
            label = "Name",
            value = applicationData.fullName
        )

        ReviewDetail(
            label = "IC No.",
            value = applicationData.icNumber
        )

        ReviewDetail(
            label = "Date of Birth",
            value = applicationData.dateOfBirth
        )

        ReviewDetail(
            label = "Email",
            value = applicationData.email
        )

        ReviewDetail(
            label = "Mobile No.",
            value =
                "${applicationData.phoneCountryCode} ${applicationData.phoneNumber}"
        )

        ReviewDetail(
            label = "Name on Card",
            value = applicationData.nameOnCard
        )

        ReviewDetail(
            label = "Education",
            value = applicationData.education
        )

        ReviewDetail(
            label = "Gender",
            value = applicationData.gender
        )

        ReviewDetail(
            label = "Race",
            value = applicationData.race
        )

        ReviewDetail(
            label = "Marital Status",
            value = applicationData.maritalStatus,
            addBottomSpacing = false
        )
    }
}

@Composable
private fun JobDetailsReviewCard(
    applicationData: CreditCardApplicationData,
    onEdit: () -> Unit
) {

    val shape = RoundedCornerShape(12.dp)

    Column(
        modifier = Modifier
            .fillMaxWidth()
            .shadow(
                elevation = 3.dp,
                shape = shape
            )
            .background(
                color = White,
                shape = shape
            )
            .border(
                width = 1.dp,
                color = Color(0xFFEEEEEE),
                shape = shape
            )
            .padding(17.dp)
    ) {

        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement =
                Arrangement.SpaceBetween,
            verticalAlignment =
                Alignment.CenterVertically
        ) {

            Text(
                text = "Job Details",
                color = Color.Black,
                fontSize = 14.sp,
                fontWeight = FontWeight.Bold
            )

            Text(
                text = "EDIT",
                modifier = Modifier.clickable {
                    onEdit()
                },
                color = PrimaryBlue,
                fontSize = 12.sp,
                letterSpacing = 0.6.sp
            )
        }

        Spacer(
            modifier = Modifier.height(9.dp)
        )

        Box(
            modifier = Modifier
                .fillMaxWidth()
                .height(1.dp)
                .background(
                    Color(0xFFF2F4F6)
                )
        )

        Spacer(
            modifier = Modifier.height(8.dp)
        )

        ReviewDetail(
            label = "Employer Name",
            value = applicationData.employerName
        )

        ReviewDetail(
            label = "Occupation",
            value = applicationData.occupation
        )

        ReviewDetail(
            label = "Sector",
            value = applicationData.sector
        )

        ReviewDetail(
            label = "Employment Type",
            value = applicationData.employmentType
        )

        ReviewDetail(
            label = "Business Classification",
            value = applicationData.businessClassification
        )

        ReviewDetail(
            label = "Length of Service",
            value =
                "${applicationData.yearsOfService} years " +
                        "${applicationData.monthsOfService} months",
            addBottomSpacing = false
        )
    }
}

@Composable
private fun ExtraDetailsReviewCard(
    applicationData: CreditCardApplicationData,
    onEdit: () -> Unit
) {

    val shape = RoundedCornerShape(12.dp)

    Column(
        modifier = Modifier
            .fillMaxWidth()
            .shadow(
                elevation = 3.dp,
                shape = shape
            )
            .background(
                color = White,
                shape = shape
            )
            .border(
                width = 1.dp,
                color = Color(0xFFEEEEEE),
                shape = shape
            )
            .padding(17.dp)
    ) {

        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement =
                Arrangement.SpaceBetween,
            verticalAlignment =
                Alignment.CenterVertically
        ) {

            Text(
                text = "Extra Details",
                color = Color.Black,
                fontSize = 14.sp,
                fontWeight = FontWeight.Bold
            )

            Text(
                text = "EDIT",
                modifier = Modifier.clickable {
                    onEdit()
                },
                color = PrimaryBlue,
                fontSize = 12.sp,
                letterSpacing = 0.6.sp
            )
        }

        Spacer(
            modifier = Modifier.height(9.dp)
        )

        Box(
            modifier = Modifier
                .fillMaxWidth()
                .height(1.dp)
                .background(
                    Color(0xFFF2F4F6)
                )
        )

        Spacer(
            modifier = Modifier.height(8.dp)
        )

        ReviewDetail(
            label = "Monthly Net Income",
            value = "RM ${applicationData.monthlyNetIncome}"
        )

        ReviewDetail(
            label = "Monthly Commitments",
            value = "RM ${applicationData.monthlyCommitments}"
        )

        ReviewDetail(
            label = "Retirement Income Source",
            value = applicationData.retirementIncomeSource
        )

        ReviewDetail(
            label = "Statement Delivery",
            value = applicationData.statementDelivery
        )

        ReviewDetail(
            label = "Collection State",
            value = applicationData.collectionState
        )

        ReviewDetail(
            label = "Collection District",
            value = applicationData.collectionDistrict
        )

        ReviewDetail(
            label = "Collection Branch",
            value = applicationData.collectionBranch,
            addBottomSpacing = false
        )
    }
}

@Composable
private fun ReviewDetail(
    label: String,
    value: String,
    addBottomSpacing: Boolean = true
) {

    Column(
        modifier =
            Modifier.fillMaxWidth()
    ) {

        Text(
            text = label,

            color =
                Color(0xFF666666),

            fontSize = 13.sp,

            lineHeight = 20.sp
        )


        Text(
            text = value,

            color =
                Color(0xFF191C1E),

            fontSize = 14.sp,

            lineHeight = 20.sp
        )


        if (addBottomSpacing) {

            Spacer(
                modifier =
                    Modifier.height(12.dp)
            )
        }
    }
}

File: unknownbank_flutter/lib/screens/credit_card_review_screen.dart
import 'package:flutter/material.dart';

import '../models/credit_card_application_data.dart';
import '../theme/app_colors.dart';

import 'credit_card_personal_details_screen.dart';
import 'credit_card_about_you_screen.dart';
import 'credit_card_job_details_screen.dart';
import 'credit_card_extra_details_screen.dart';

class CreditCardReviewScreen extends StatefulWidget {
  final String productName;
  final CreditCardApplicationData applicationData;

  final ValueChanged<CreditCardApplicationData>
      onApplicationDataChange;

  final VoidCallback onNext;

  const CreditCardReviewScreen({
    super.key,
    required this.productName,
    required this.applicationData,
    required this.onApplicationDataChange,
    required this.onNext,
  });

  @override
  State<CreditCardReviewScreen> createState() =>
      _CreditCardReviewScreenState();
}

class _CreditCardReviewScreenState
    extends State<CreditCardReviewScreen> {
  late CreditCardApplicationData data;

  @override
  void initState() {
    super.initState();
    data = widget.applicationData;
  }

  void _update(
    CreditCardApplicationData updatedData,
  ) {
    setState(() {
      data = updatedData;
    });

    widget.onApplicationDataChange(updatedData);
  }

  Future<void> _editPersonal() async {
    await Navigator.of(context).push(
      MaterialPageRoute(
        builder: (personalContext) =>
            CreditCardPersonalDetailsScreen(
          productName: widget.productName,
          applicationData: data,

          onApplicationDataChange: _update,

          onNext: () {
            Navigator.of(personalContext).push(
              MaterialPageRoute(
                builder: (aboutContext) =>
                    CreditCardAboutYouScreen(
                  productName:
                      widget.productName,
                  applicationData: data,

                  onApplicationDataChange:
                      _update,

                  onNext: () {
                    // About You -> Review
                    Navigator.of(aboutContext).pop();

                    // Personal Details -> Review
                    Navigator.of(personalContext).pop();
                  },
                ),
              ),
            );
          },
        ),
      ),
    );

    if (mounted) {
      setState(() {});
    }
  }

  Future<void> _editJob() async {
    await Navigator.of(context).push(
      MaterialPageRoute(
        builder: (jobContext) =>
            CreditCardJobDetailsScreen(
          productName: widget.productName,
          applicationData: data,

          onApplicationDataChange: _update,

          onNext: () {
            Navigator.of(jobContext).pop();
          },
        ),
      ),
    );

    if (mounted) {
      setState(() {});
    }
  }

  Future<void> _editExtra() async {
    await Navigator.of(context).push(
      MaterialPageRoute(
        builder: (extraContext) =>
            CreditCardExtraDetailsScreen(
          productName: widget.productName,
          applicationData: data,

          onApplicationDataChange: _update,

          onNext: () {
            Navigator.of(extraContext).pop();
          },
        ),
      ),
    );

    if (mounted) {
      setState(() {});
    }
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Stack(
        children: [
          Positioned.fill(
            child: Image.asset(
              'assets/images/full_background_light.png',
              fit: BoxFit.cover,
            ),
          ),

          SafeArea(
            child: Column(
              children: [
                SizedBox(
                  height: 56,
                  child: Stack(
                    alignment: Alignment.center,
                    children: [
                      Align(
                        alignment: Alignment.centerLeft,
                        child: Padding(
                          padding:
                              const EdgeInsets.only(
                            left: 8,
                          ),
                          child: IconButton(
                            onPressed: () {
                              Navigator.of(context).pop();
                            },
                            icon: const Icon(
                              Icons.arrow_back,
                              color:
                                  AppColors.textPrimary,
                            ),
                          ),
                        ),
                      ),

                      Padding(
                        padding: const EdgeInsets.only(
                          left: 55,
                          right: 30,
                        ),
                        child: Text(
                          'Applying for ${widget.productName}',
                          maxLines: 1,
                          overflow:
                              TextOverflow.ellipsis,
                          style: const TextStyle(
                            fontSize: 13,
                            fontWeight:
                                FontWeight.bold,
                            color:
                                Color(0xFF333333),
                          ),
                        ),
                      ),
                    ],
                  ),
                ),

                const Padding(
                  padding: EdgeInsets.fromLTRB(
                    20,
                    20,
                    23,
                    0,
                  ),
                  child: Row(
                    mainAxisAlignment:
                        MainAxisAlignment
                            .spaceBetween,
                    children: [
                      Text(
                        'Review Details',
                        style: TextStyle(
                          fontSize: 15,
                          fontWeight:
                              FontWeight.bold,
                          color:
                              Color(0xFF333333),
                        ),
                      ),

                      Text(
                        '100%',
                        style: TextStyle(
                          fontSize: 14,
                          fontWeight:
                              FontWeight.w500,
                          color:
                              Color(0xFF0EAF5F),
                        ),
                      ),
                    ],
                  ),
                ),

                const SizedBox(height: 11),

                Padding(
                  padding:
                      const EdgeInsets.symmetric(
                    horizontal: 20,
                  ),
                  child: Container(
                    height: 6,
                    decoration: BoxDecoration(
                      color:
                          const Color(0xFF0EAF5F),
                      borderRadius:
                          BorderRadius.circular(
                        100,
                      ),
                    ),
                  ),
                ),

                const SizedBox(height: 26),

                Expanded(
                  child: SingleChildScrollView(
                    padding:
                        const EdgeInsets.fromLTRB(
                      20,
                      0,
                      20,
                      110,
                    ),
                    child: Column(
                      children: [
                        _ReviewCard(
                          title:
                              'Personal Details',
                          showPersonIcon: true,
                          showEditIcon: true,
                          onEdit: _editPersonal,
                          details: [
                            MapEntry(
                              'Title',
                              data.title,
                            ),
                            MapEntry(
                              'Name',
                              data.fullName,
                            ),
                            MapEntry(
                              'IC No.',
                              data.icNumber,
                            ),
                            MapEntry(
                              'Date of Birth',
                              data.dateOfBirth,
                            ),
                            MapEntry(
                              'Email',
                              data.email,
                            ),
                            MapEntry(
                              'Mobile No.',
                              '${data.phoneCountryCode} ${data.phoneNumber}',
                            ),
                            MapEntry(
                              'Name on Card',
                              data.nameOnCard,
                            ),
                            MapEntry(
                              'Education',
                              data.education,
                            ),
                            MapEntry(
                              'Gender',
                              data.gender,
                            ),
                            MapEntry(
                              'Race',
                              data.race,
                            ),
                            MapEntry(
                              'Marital Status',
                              data.maritalStatus,
                            ),
                            MapEntry(
                              "Mother's Name",
                              data.motherName,
                            ),
                          ],
                        ),

                        const SizedBox(height: 16),

                        _ReviewCard(
                          title: 'Job Details',
                          onEdit: _editJob,
                          details: [
                            MapEntry(
                              'Employer Name',
                              data.employerName,
                            ),
                            MapEntry(
                              'Occupation',
                              data.occupation,
                            ),
                            MapEntry(
                              'Sector',
                              data.sector,
                            ),
                            MapEntry(
                              'Employment Type',
                              data.employmentType,
                            ),
                            MapEntry(
                              'Business Classification',
                              data
                                  .businessClassification,
                            ),
                            MapEntry(
                              'Length of Service',
                              '${data.yearsOfService} years '
                                  '${data.monthsOfService} months',
                            ),
                          ],
                        ),

                        const SizedBox(height: 16),

                        _ReviewCard(
                          title: 'Extra Details',
                          onEdit: _editExtra,
                          details: [
                            MapEntry(
                              'Monthly Net Income',
                              'RM ${data.monthlyNetIncome}',
                            ),
                            MapEntry(
                              'Monthly Commitments',
                              'RM ${data.monthlyCommitments}',
                            ),
                            MapEntry(
                              'Retirement Income Source',
                              data
                                  .retirementIncomeSource,
                            ),
                            MapEntry(
                              'Statement Delivery',
                              data.statementDelivery,
                            ),
                            MapEntry(
                              'Collection State',
                              data.collectionState,
                            ),
                            MapEntry(
                              'Collection District',
                              data.collectionDistrict,
                            ),
                            MapEntry(
                              'Collection Branch',
                              data.collectionBranch,
                            ),
                          ],
                        ),
                      ],
                    ),
                  ),
                ),
              ],
            ),
          ),

          Positioned(
            left: 16,
            right: 16,
            bottom: 16,
            child: SafeArea(
              top: false,
              child: SizedBox(
                height: 46,
                child: FilledButton(
                  onPressed: widget.onNext,
                  style: FilledButton.styleFrom(
                    backgroundColor:
                        AppColors.primary,
                    shape:
                        RoundedRectangleBorder(
                      borderRadius:
                          BorderRadius.circular(
                        100,
                      ),
                    ),
                  ),
                  child: const Text(
                    'Next',
                    style: TextStyle(
                      fontSize: 15,
                      fontWeight:
                          FontWeight.bold,
                    ),
                  ),
                ),
              ),
            ),
          ),
        ],
      ),
    );
  }
}

class _ReviewCard extends StatelessWidget {
  final String title;
  final List<MapEntry<String, String>> details;
  final VoidCallback onEdit;

  final bool showPersonIcon;
  final bool showEditIcon;

  const _ReviewCard({
    required this.title,
    required this.details,
    required this.onEdit,
    this.showPersonIcon = false,
    this.showEditIcon = false,
  });

  @override
  Widget build(BuildContext context) {
    return Container(
      width: double.infinity,
      padding: const EdgeInsets.all(17),
      decoration: BoxDecoration(
        color: Colors.white,
        borderRadius:
            BorderRadius.circular(12),
        border: Border.all(
          color: const Color(0xFFEEEEEE),
        ),
        boxShadow: const [
          BoxShadow(
            color: Color(0x22000000),
            blurRadius: 6,
            offset: Offset(0, 2),
          ),
        ],
      ),
      child: Column(
        crossAxisAlignment:
            CrossAxisAlignment.start,
        children: [
          Row(
            children: [
              if (showPersonIcon) ...[
                Image.asset(
                  'assets/images/review_person.png',
                  width: 16,
                  height: 16,
                ),

                const SizedBox(width: 8),
              ],

              Expanded(
                child: Text(
                  title,
                  style: const TextStyle(
                    fontSize: 14,
                    fontWeight:
                        FontWeight.bold,
                    color: Colors.black,
                  ),
                ),
              ),

              InkWell(
                onTap: onEdit,
                child: Row(
                  children: [
                    const Text(
                      'EDIT',
                      style: TextStyle(
                        fontSize: 12,
                        letterSpacing: 0.6,
                        color:
                            AppColors.primary,
                      ),
                    ),

                    if (showEditIcon) ...[
                      const SizedBox(width: 4),

                      Image.asset(
                        'assets/images/review_edit.png',
                        width: 12,
                        height: 12,
                      ),
                    ],
                  ],
                ),
              ),
            ],
          ),

          const SizedBox(height: 9),

          const Divider(
            height: 1,
            color: Color(0xFFF2F4F6),
          ),

          const SizedBox(height: 8),

          for (int i = 0;
              i < details.length;
              i++)
            _ReviewDetail(
              label: details[i].key,
              value: details[i].value,
              addBottomSpacing:
                  i != details.length - 1,
            ),
        ],
      ),
    );
  }
}

class _ReviewDetail extends StatelessWidget {
  final String label;
  final String value;
  final bool addBottomSpacing;

  const _ReviewDetail({
    required this.label,
    required this.value,
    required this.addBottomSpacing,
  });

  @override
  Widget build(BuildContext context) {
    return Column(
      crossAxisAlignment:
          CrossAxisAlignment.start,
      children: [
        Text(
          label,
          style: const TextStyle(
            fontSize: 13,
            height: 1.54,
            color: Color(0xFF666666),
          ),
        ),

        Text(
          value,
          style: const TextStyle(
            fontSize: 14,
            height: 1.43,
            color: Color(0xFF191C1E),
          ),
        ),

        if (addBottomSpacing)
          const SizedBox(height: 12),
      ],
    );
  }
}